benches/centralized_telescope/number_steps.rs

```diff
@@ -0,0 +1,72 @@
+//! Benchmarking the number of DFS calls, aka number of steps, of the
+//! Centralized Telescope scheme
+
+use criterion::{black_box, criterion_group, criterion_main, Criterion};
+use rand_chacha::ChaCha20Rng;
+use rand_core::SeedableRng;
+use std::time::Duration;
+
+mod utils;
+use utils::{
+    common::{
+        criterion_helpers::{
+            centralized::{benchmarks, BenchParam},
+            Steps,
+        },
+        test_vectors::centralized::SHORT_TESTS,
+    },
+    setup, NAME,
+};
+
+use alba::centralized_telescope::proof::Proof;
+
+/// Function benchmarking `sample_size` times the number of DFS calls, aka steps
+#[allow(clippy::unit_arg)]
+fn prove_steps(param: &BenchParam, truncate_size: u64, n: u64) -> u64 {
+    let mut rng = ChaCha20Rng::from_entropy();
+    let mut total_steps = 0u64;
+
+    // Setup
+    let (mut dataset, telescope) = setup(&mut rng, param);
+    let set_size = telescope.get_set_size();
+    let params = telescope.get_params();
+    // Truncate the dataset to give truncate_size elements to the prover
+    dataset.truncate(truncate_size as usize);
+
+    // Iterate on each sample `n` times
+    for _ in 0..n {
+        // Bench the number of steps/DFS calls while generating a proof
+        black_box({
+            let steps = Proof::bench(set_size, &params, &dataset).0;
```

Comment:
Since the number of steps should be the same in all iterations, we could call `bench` once and multiply the result by `n` to optimize running time.

Reply:
Good call! Done